Task states from registry

JanWehrmann opened the pull request.

Commit:
add functions to get STATE_ID + tests

saarnold commented on src/TypeRegistry.cpp:
```diff
@@ -106,4 +171,15 @@ bool TypeRegistry::getTypekitDefiningType(const std::string& typeName, std::stri
 
-}
\ No newline at end of file
+bool TypeRegistry::getStateID(const std::string &task_model_name, std::string &state_name, unsigned& id) const
```
`state_name` should be a const reference

saarnold commented:
Looks already very good :+1:

saarnold commented on test/test_typeregistry.cpp:
```diff
@@ -0,0 +1,69 @@
+    Fixture()
+    {
+        type_registries.loadStatesFromCustomPath("../../test/testfile.tlb");
+        type_registries.loadTypesFromCustomPath("../../test/testfile.typelist", "auv_control");
```
The test files must be installed as well. The path of the build folder isn't fixed.

One solution would be to add
```
configure_file(${CMAKE_CURRENT_SOURCE_DIR}/testfile.tlb
               ${CMAKE_CURRENT_BINARY_DIR}/testfile.tlb COPYONLY)
```
and assume that the test files are in the same folder than the test executable.

JanWehrmann committed:
Extended TypeRegistry - second iteration.

Merged loadTypelist and loadTypeRegistries into one
function.

Transferred generating the mapping to seperate
functions.

Fixed one memory leak and some minor issues.

Added dedicated testfiles.
